src/Microsoft.AspNetCore.Sockets.Http/Internal/ConnectionLogScope.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Sockets.Internal
+{
+    public class ConnectionLogScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private string _connectionId;
+
+        private readonly HttpContext _context;
+        private string _cachedToString;
+
+        public ConnectionLogScope(HttpContext context)
+        {
+            _context = context;
+        }
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (Count == 1 && index == 0)
+                {
+                    return new KeyValuePair<string, object>("SocketsConnectionId", ConnectionId);
+                }
+
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+        }
+
+        public int Count => string.IsNullOrEmpty(ConnectionId) ? 0 : 1;
+
+        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+        {
+            for (var i = 0; i < Count; ++i)
+            {
+                yield return this[i];
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public override string ToString()
+        {
+            if (_cachedToString == null)
+            {
+                if (!string.IsNullOrEmpty(ConnectionId))
+                {
+                    _cachedToString = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "SocketsConnectionId:{0}",
+                        ConnectionId);
+                }
+            }
+
+            return _cachedToString;
+        }
+
+        private string ConnectionId
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_connectionId))
+                {
+                    _connectionId = _context.Request.Query["id"];
```

Comment:
I was thinking about any potential concurrency issues with this. This might potentially access the `HttpContext` on different threads (in parallel) right? Specifically here https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs#L218. The saving grace is likely that the feature is already populated *and* the value is likely stored before then. We should put a comment in here....